### PR TITLE
api docs for tsdataset 

### PR DESCRIPTION
**Pull request by cabuliwallah**
You can preview the doc at [here](https://yuyin-analytics-zoo.readthedocs.io/en/latest/doc/PythonAPI/Chronos/tsdataset.html)

### CODE REVIEW

File: docs/readthedocs/source/doc/PythonAPI/Chronos/tsdataset.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+TSDataset
+===========
+
+chronos.data.tsdataset
+----------------------------------------
+
+TSDataset is an abstract of time series dataset.
```
Comment (TheaperDeng) [inline docs/readthedocs/source/doc/PythonAPI/Chronos/tsdataset.rst:7]: Time series data is a special data formulation with specific operations. TSDataset is an abstract of time series dataset, which provides various data processing operation (e.g. impute, deduplicate, resample, scale/unscale, roll) and feature engineering methods (e.g. datetime feature, aggregation feature). Cascade call is supported for most of the methods.
TSDataset can be initialized from a pandas dataframe and be converted to pandas dataframe or numpy ndarray.

File: pyzoo/zoo/chronos/data/tsdataset.py [hunk 2]
```diff
@@ -87,7 +87,7 @@ def from_pandas(df,
         :param id_col: (optional) a str indicates the col name of dataframe id.
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/data/tsdataset.py:87]: Line 80: Initialize tsdataset(s) from pandas dataframe.

Comment (TheaperDeng) [reply pyzoo/zoo/chronos/data/tsdataset.py:87]: id_col: (optional) a str indicates the col name of dataframe id. If it is not explicitly stated, then the data is interpreted as only containing a single id.

File: pyzoo/zoo/chronos/data/tsdataset.py [hunk 3]
```diff
@@ -102,17 +102,14 @@ def from_pandas(df,
...
-        01        2019-01-02    2.6     0                   2
-
         Create a tsdataset instance by:
 
+        >>> #Here is a df example:
+        >>> #id        datetime      value   "extra feature 1"   "extra feature 2"
+        >>> #00        2019-01-01    1.9     1                   2
+        >>> #01        2019-01-01    2.3     0                   9
+        >>> #00        2019-01-02    2.4     3                   4
+        >>> #01        2019-01-02    2.6     0                   2
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/data/tsdataset.py:112]: add a space after #

File: pyzoo/zoo/chronos/data/tsdataset.py [hunk 4]
```diff
@@ -177,7 +177,7 @@ def impute(self, mode="last", const_num=0):
     def deduplicate(self):
         '''
         Remove those duplicated rows which has exactly the same values in each feature_col for
-        each multivariate timeseries distinguished by id_col
+        each multivariate timeseries distinguished by id_col.
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/data/tsdataset.py:180]: Remove those duplicated record which has exactly the same values in each feature_col for each multivariate timeseries distinguished by id_col

File: pyzoo/zoo/chronos/data/tsdataset.py [hunk 5]
```diff
@@ -151,20 +148,23 @@ def from_pandas(df,
...
+            "linear": impute by linear interpolation.
+        :param const_num:  indicates the const number to fill, which is only effective when mode
+            is set to "const".
 
         :return: the tsdataset instance.
 
-        Note: It is preferred that `impute` is called after `resample` while before
-              `roll` if needed.
+        Note: It is preferred that `impute` is called after `resample` while before `roll` if
+        needed.
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/data/tsdataset.py:167]: remove all the Note. We will add them later.

File: pyzoo/zoo/chronos/data/tsdataset.py [hunk 6]
```diff
@@ -222,7 +223,7 @@ def gen_dt_feature(self):
         '''
         Generate datetime feature for each row. Currently we generate following features:
         "MINUTE", "DAY", "DAYOFYEAR", "HOUR", "WEEKDAY", "WEEKOFYEAR", "MONTH", "IS_AWAKE",
-        "IS_BUSY_HOURS", "IS_WEEKEND"
+        "IS_BUSY_HOURS", "IS_WEEKEND".
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/data/tsdataset.py:226]: add simple description to all features here. e.g.
"MINUTE": The minute of the time stamp.
...

File: pyzoo/zoo/chronos/data/tsdataset.py [hunk 7]
```diff
@@ -442,22 +443,32 @@ def to_numpy(self):
...
+
         :return: the tsdataset instance.
+
+        Assume there is a training set tsdata and a test set tsdata_test.
+        scale() should be called first on training set with default value fit=True,
+        then be called on test set with the same scaler and fit=False.
+
+        >>> from sklearn.preprocessing import StandardScaler
+        >>> scaler = StandardScaler()
+        >>> tsdata.scale(scaler)
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/data/tsdataset.py:470]: tsdata_train.scale(scaler, fit=True)

**Comment by TheaperDeng** [comment]
pls run jenkins

**Comment by cabuliwallah** [comment]
Jenkins passed:
http://10.239.176.111:18888/job/ZOO-PR-Validation/5815/

**Pull request merged**